### PR TITLE
Scan first 64 blocks for an HFS partition

### PR DESCRIPTION
**Pull request by erichelgeson**
Tested this with a BlueSCSI image that contains a Partition Map, SCSI Driver, and HFS Partition
Tested with DC42 image
Tested with Hypercard 2.4.1 iso file

Screenshot 2023-09-19 at 4 40 43 PM

### CODE REVIEW

File: src/SONYEMDV.c [hunk 1]
```diff
@@ -438,16 +440,21 @@ LOCALFUNC tMacErr vSonyNextPendingInsert(tDrive *Drive_No)
 						}
 					}
 
-					// Handle .iso's with HFS partitions. Based on the Basilisk II find_hfs_partition implementation.
+					// Handle file's with HFS partitions. Based on the Basilisk II find_hfs_partition implementation.
```
Comment (mihaip) [inline src/SONYEMDV.c]: Nit: typo, should be "files"

File: src/SONYEMDV.c [hunk 2]
```diff
@@ -438,16 +440,21 @@ LOCALFUNC tMacErr vSonyNextPendingInsert(tDrive *Drive_No)
...
-							&Temp[0x400]);
-						if (drSigWord == 0x504d) { // HFS partition map magic number.
-							ui3p map = &Temp[0x400];
-							if (strcmp((char *)(map + 48), "Apple_HFS") == 0) {
-								DataOffset = ((map[8] << 24) | (map[9] << 16) | (map[10] << 8) | map[11]) << 9;
-								DataSize = 512 * ((map[12] << 24) | (map[13] << 16) | (map[14] << 8) | map[15]);
-								gotFormat = trueblnr;
+						int i;
+						for (i = 0; i < checkheaderblocks; i++)
+						{
```
Comment (mihaip) [inline src/SONYEMDV.c]: Nit: can you be consistent with the existing code as far as opening brace placement (it goes on the previous line).

File: src/SONYEMDV.c [hunk 3]
```diff
@@ -438,16 +440,21 @@ LOCALFUNC tMacErr vSonyNextPendingInsert(tDrive *Drive_No)
...
 					if (! gotFormat) {
-						ui4r drSigWord = do_get_mem_word(
-							&Temp[0x400]);
-						if (drSigWord == 0x504d) { // HFS partition map magic number.
-							ui3p map = &Temp[0x400];
-							if (strcmp((char *)(map + 48), "Apple_HFS") == 0) {
-								DataOffset = ((map[8] << 24) | (map[9] << 16) | (map[10] << 8) | map[11]) << 9;
-								DataSize = 512 * ((map[12] << 24) | (map[13] << 16) | (map[14] << 8) | map[15]);
-								gotFormat = trueblnr;
+						int i;
```
Comment (mihaip) [inline src/SONYEMDV.c:445]: Nit: could be done inline in the `for` loop.

**Review by mihaip — COMMENTED** [review]
Just some minor style feedback.

File: src/SONYEMDV.c (hunk 3 above)
Comment (erichelgeson) [reply src/SONYEMDV.c:445]: This was following the style of the other for loops in the program.

**Pull request merged**